src/vs/workbench/contrib/scm/browser/dirtydiffDecorator.ts

```diff
@@ -1067,37 +1067,10 @@ function compareChanges(a: IChange, b: IChange): number {
 	return a.originalEndLineNumber - b.originalEndLineNumber;
 }
 
-export function createProviderComparer(uri: URI): (a: ISCMProvider, b: ISCMProvider) => number {
-	return (a, b) => {
-		const aIsParent = isEqualOrParent(uri, a.rootUri!);
-		const bIsParent = isEqualOrParent(uri, b.rootUri!);
-
-		if (aIsParent && bIsParent) {
-			return a.rootUri!.fsPath.length - b.rootUri!.fsPath.length;
-		} else if (aIsParent) {
-			return -1;
-		} else if (bIsParent) {
-			return 1;
-		} else {
-			return 0;
-		}
-	};
-}
-
-export async function getOriginalResource(scmService: ISCMService, uri: URI): Promise<URI | null> {
-	const providers = Iterable.map(scmService.repositories, r => r.provider);
-	const rootedProviders = Array.from(Iterable.filter(providers, p => !!p.rootUri));
-
-	rootedProviders.sort(createProviderComparer(uri));
-
-	const result = await first(rootedProviders.map(p => () => p.getOriginalResource(uri)));
-
-	if (result) {
-		return result;
-	}
 
-	const nonRootedProviders = Iterable.filter(providers, p => !p.rootUri);
-	return first(Array.from(nonRootedProviders, p => () => p.getOriginalResource(uri)));
+export async function getOriginalResource(quickDiffService: IQuickDiffService, uri: URI): Promise<URI | null> {
+	const quickDiffs = await quickDiffService.getQuickDiffs(uri);
+	return quickDiffs ? quickDiffs[0].originalResource : null;
```

Comment:
This should check for `length > 0` instead of the `undefined`, since it won't ever be `undefined` (check below).